### PR TITLE
First Pass at Building the Docs Site

### PR DESCRIPTION
**Pull request by runewake2**
This is an initial pass at creating a razor-press site to provide a landing page, blog and documentation about the project.

## Details
- This is a trimmed down version of the Razor-Press template
- `Letterbook.Docs` is where the project lives, the `Readme` describes how to build and run the site.
- I've added a custom container to embedding Mastodon posts into articles.
- The blog and docs portions still need work, they are present but not really functional.
  - blog articles only work on `/blog` right now, I need to get this working on subpages.
  - docs page doesn't 404, but is empty - would love input on how to make that look.
- The GH Pages publishing is implemented, but I need to do more work to test this still - idk how it works with razor-press' design.
- There is still a lot of templated code I would like to clean up from the original template.

## Related
- #258

### CODE REVIEW

File: Letterbook.Docs/_pages/what-is-letterbook.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+title: What is Letterbook
+---
+
+Letterbook is a federated microblogging service, implementing ActivityPub. The goal for the project is to make hosting a fediverse server a better and more sustainable experience. We also want to make it a safer and more inclusive space for the people who make the fediverse their social media home. Those are big aspirations, but we have some solid plans about how to start, and we would love more input about where to go from there.
+
+::: Mastodon https://tech.lgbt/@jenniferplusplus/110567090111682149
```
Comment (jenniferplusplus) [inline Letterbook.Docs/_pages/what-is-letterbook.md:7]: ooh, is this the syntax to embed a mastodon post? That is :fire:

File: Letterbook.Docs/_pages/deploy.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+title: Deploy Letterbook
+---
+
+Something about docker compose
```
Comment (jenniferplusplus) [inline Letterbook.Docs/_pages/deploy.md:5]: tbh, this is far enough in the future that it's probably best to just not mention it yet.

I expect docker compose will be one option, but not the one I would recommend to most people. It's not all that difficult to host apt and rpm repositories, and I expect to make those the preferred install and deploy path.

File: Letterbook.Docs/_pages/index.md
Comment (jenniferplusplus) [inline Letterbook.Docs/_pages/index.md:1]: oh let's definitely get rid of this content

File: Letterbook.Docs/_videos/projects/razor-press.md
Comment (jenniferplusplus) [inline Letterbook.Docs/_videos/projects/razor-press.md:1]: And let's get rid of these razor press video pages. We can include videos as a part of the readme guide

File: .github/workflows/publish-web.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+              echo "Persist CNAME '$custom_domain' to ./Letterbook.Docs/wwwroot/CNAME"
+              echo "$custom_domain" >> ./Letterbook.Docs/wwwroot/CNAME
+              echo "custom_domain=$custom_domain" >> $GITHUB_ENV
+            fi
+          fi
+
+      - name: Build tailwind
+        working-directory: ./Letterbook.Docs
+        run: |
+          npm run build
```
[the remark this answers is not in the document]
Comment (runewake2) [reply .github/workflows/publish-web.yml:60]: hmm, let me take a look at how to do that, the default template is kind of big and messy so I've been slowly pulling bits out. I suspect that might be a bit weird to get working, but we can see.

File: Letterbook.Docs/_pages/what-is-letterbook.md (hunk 1 above)
Comment (runewake2) [reply Letterbook.Docs/_pages/what-is-letterbook.md:7]: Yeah, I added a mastodon embed feature, seemed handy.  Mastodon uses an embed script (ex: https://tech.lgbt/embed.js, https://hachyderm.io/embed.js) and I think I'll pull that into the project itself to simplify things a bit - but it's kind of messy. The embed script is slightly different across servers (though I don't think significantly so). It also probably only works with Mastodon clients right now - Sharkey won't work for example.

File: .github/workflows/publish-web.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jenniferplusplus) [reply .github/workflows/publish-web.yml:60]: You're welcome to have a look if you want to, but it doesn't have to be now. I'd rather get some docs up and improve it over time.

[the remark this answers is not in the document]
Comment (jenniferplusplus) [reply .github/workflows/publish-web.yml:60]: My thinking has been to use MS library manager to handle web dependencies. You can see that a little bit in this branch:
https://github.com/Letterbook/Letterbook/compare/enhancement/web-assets?expand=1#diff-ce948244669fb9aeff7e205a04bd1fee2b7063896cd5eca74272be48fa883fa6

That gets just the pico.min.css file and license from the picocss package, so we don't need npm. That process should work for any package hosted on npm that includes pre-built assets

**Comment by jenniferplusplus** [comment]
fwiw, I just pulled this and fired it up. It worked right out of the box, and seems great

**Comment by runewake2** [comment]
Great! It'll be a few days until I can finish cleaning this up, it's Pride this weekend but I should have it ready to go early next week.

**Comment by jenniferplusplus** [comment]
@runewake2 just to follow up: is this still in progress? Is there anything I can help with?

**Pull request merged**